src/module_type_translator.ts

```diff
@@ -579,3 +579,22 @@ export class ModuleTypeTranslator {
     };
   }
 }
+
+/** Mutually exclusive visibility jsdoc tags. */
+export type Visibility = 'public'|'private'|'protected'|'export';
```

Comment:
`'export'` is really not a visibility in TS, and it isn't in JS either - I think it's just a hack/special cse that `@export` maps to `public`.

As such, I think this change adds quite a bunch of confusion to the code base here. E.g. adding `@export` as a visibility is super surprising to me.

I think I also don't really follow on the premise. If a property is used by your template, it is used outside of your lexical scope, and thus it must not be `private`. Using `protected` similarly is a hack - what really happens here is that your property is used by code outside of the current scope that you have no inheritance relationship with. This question comes up occasionally for frameworks that sneakily use reflection to access properties. We've so far decided (and documented) that TS code visibility should match the actual use, i.e. be public.

With that in mind, I'd consider adding a check here and reporting it as an error if a property has the magic `@Export` decorator on it and isn't `public`. WDYT?

Reply:
`'export'` is effectively a visibility to Closure Compiler. It implies `public` with additional semantics, and it's mutually exclusive to the other visibilities. This was super surprising to me too, but what else do you call such a thing but a visibility? I'll rename to ClosureVisibility, and add some more docs.

Polymer templates behave as you describe, but lit-html templates do not. There, expressions in templates are ordinary JS expressions inside of a tagged template literal, so they follow scoping and visibility. I initially wrote an `@ExportDecoratedItemsIfPublic` decorator modifier which we'd only use with lit-html, but the TypeScript decorator output includes the field names as strings which we use to register getters and setters. This meant that renaming the private and protected members would break at runtime.

Long term I'd like to have a TypeScript transform that pre-processes the decorators, in which case the code would be rename safe. The other option would be to modify TypeScript's decorator output to add calls to JSCompiler_renameProperty or goog.reflect.objectProperty to make them renaming safe.

This change seems like a step in the right direction, and much simpler than those options. If you're uncomfortable with Polymer templates accessing private or protected fields then we could have one decorator modifier to use with Polymer that errors on protected and private and another for use with lit that does not.